datafusion/substrait/src/physical_plan/consumer.rs

```diff
@@ -0,0 +1,144 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+use async_recursion::async_recursion;
+use chrono::DateTime;
+use datafusion::arrow::datatypes::Schema;
+use datafusion::datasource::listing::PartitionedFile;
+use datafusion::datasource::object_store::ObjectStoreUrl;
+use datafusion::error::{DataFusionError, Result};
+use datafusion::physical_plan::file_format::{FileScanConfig, ParquetExec};
+use datafusion::physical_plan::ExecutionPlan;
+use datafusion::prelude::SessionContext;
+use object_store::ObjectMeta;
+use std::collections::HashMap;
+use std::sync::Arc;
+use substrait::proto::read_rel::local_files::file_or_files::PathType;
+use substrait::proto::{
+    expression::MaskExpression, read_rel::ReadType, rel::RelType, Rel,
+};
+
+/// Convert Substrait Rel to DataFusion LogicalPlan
+#[async_recursion]
+pub async fn from_substrait_rel(
+    _ctx: &mut SessionContext,
+    rel: &Rel,
+    _extensions: &HashMap<u32, &String>,
+) -> Result<Arc<dyn ExecutionPlan>> {
+    match &rel.rel_type {
+        Some(RelType::Read(read)) => {
+            if read.filter.is_some() || read.best_effort_filter.is_some() {
+                return Err(DataFusionError::NotImplemented(
```

Comment:
Is this just because this is a first pass at getting the framework together or is it truly not possible with DataFusion today?

Reply:
Yes, just a first pass. I plan on following up with smaller PRs to make this usable.